helm/hpcc/templates/_warnings.tpl

```diff
@@ -52,4 +52,78 @@ Pass in dict with root and warnings
   {{- $_ := set $warning "msg" (printf "The configuration contains ephemeral planes: %s" (print (uniq $match.ephemeral)) )  -}}
   {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
  {{- end -}}
+ {{- /* Warn if resources not defined for any cost component or default cpuRate being used */ -}}
+ {{- if eq .root.Values.global.cost.perCpu 0.0565000000001 -}}
+  {{- $_ := set $ctx "usingDefaultCpuCost" "true" -}}
+ {{- end -}}
+ {{- $_ := set $ctx "missingResources" list -}}
+ {{- $_ := set $ctx "defaultCpuRateComponents" list -}}
+ {{- range $component := .root.Values.eclagent -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources := include "hpcc.hasResources" (dict "resources" $component.resources) -}}
+   {{- if not $hasResources -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if and $ctx.usingDefaultCpuCost (not $component.cost) -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.eclccserver -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources := include "hpcc.hasResources" (dict "resources" $component.resources) -}}
+   {{- if not $hasResources -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if and $ctx.usingDefaultCpuCost (not $component.cost) -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.thor -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources1 := include "hpcc.hasResources" (dict "resources" $component.managerResources) -}}
+   {{- $hasResources2 := include "hpcc.hasResources" (dict "resources" $component.workerResources) -}}
+   {{- $hasResources3 := include "hpcc.hasResources" (dict "resources" $component.eclAgentResources) -}}
+   {{- if or (not $hasResources1) (or (not $hasResources2 ) (not $hasResources3)) -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if and $ctx.usingDefaultCpuCost (not $component.cost) -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- if $ctx.missingResources -}}
+  {{- $warning := dict "source" "helm" "severity" "warning" -}}
+  {{- $_ := set $warning "msg" (printf "Cost calculation requires resources to be provided for %s: %s" ((len $ctx.missingResources)| plural "component" "components") ($ctx.missingResources|toStrings)) -}}
```

Comment:
ok for now, but I we might want to make the warning more general..
it's a bad (or toy) config in general if resource haven't been defined

Reply:
There is a another jira (draft PR: https://github.com/hpcc-systems/HPCC-Platform/pull/16738) to report if resources are missing.   This jira concerns costs only.